codetransformer/decompiler.py

```diff
@@ -0,0 +1,1402 @@
+import ast
+from collections import deque
+from functools import singledispatch
+import types
+
+from toolz import complement, compose, curry
+import toolz.curried.operator as op
+
+from .code import Code, Flags
+from . import instructions as instrs
+from .utils.functional import not_a, is_a
+from .utils.immutable import immutable
+from codetransformer import a as showa, d as showd  # noqa
```

Comment:
They are, but every time I'm working on anything in this file, I end up importing these in the debugger, so I finally just put them at the top.  90% of the workflow on this for me has been "register a new function, put a breakpoint at the top, look at what it gets passed, figure out what it should emit", and being able to pprint instructions and AST is sort of essential for that.